Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/API/Beckn/Search.hs

```diff
@@ -44,26 +48,56 @@ search ::
   Id DM.Merchant ->
   SignatureAuthResult ->
   SignatureAuthResult ->
-  Search.SearchReq ->
+  -- Search.SearchReq ->
+  ByteString ->
   FlowHandler AckResponse
-search transporterId (SignatureAuthResult _ subscriber) (SignatureAuthResult _ gateway) req =
-  withFlowHandlerBecknAPI . withTransactionIdLogTag req $ do
-    logTagInfo "Search API Flow" "Reached"
-    dSearchReq <- ACL.buildSearchReq transporterId subscriber req
-    Redis.whenWithLockRedis (searchLockKey dSearchReq.messageId transporterId.getId) 60 $ do
-      merchant <- DSearch.validateRequest transporterId dSearchReq
-      fork "search request processing" $
-        Redis.whenWithLockRedis (searchProcessingLockKey dSearchReq.messageId transporterId.getId) 60 $ do
-          dSearchRes <- DSearch.handler merchant dSearchReq
-          let context = req.context
-          let callbackUrl = gateway.subscriber_url
-          void $
-            CallBAP.withCallback dSearchRes.provider Context.SEARCH OnSearch.onSearchAPI context callbackUrl $ do
-              pure $ ACL.mkOnSearchMessage dSearchRes
-    pure Ack
+search transporterId (SignatureAuthResult _ subscriber) (SignatureAuthResult _ gateway) reqBS = withFlowHandlerBecknAPI $ do
+  req <- decodeReq reqBS
+  dSearchReq <-
+    case req of
+      Left reqV1 ->
+        withTransactionIdLogTag reqV1 $ do
+          logTagInfo "Search API Flow" "Reached"
+          ACL.buildSearchReqV1 transporterId subscriber reqV1
+      Right reqV2 ->
+        withTransactionIdLogTag reqV2 $ do
+          logTagInfo "SearchV2 API Flow" "Reached"
+          ACL.buildSearchReqV2 transporterId subscriber reqV2
+
+  Redis.whenWithLockRedis (searchLockKey dSearchReq.messageId transporterId.getId) 60 $ do
+    merchant <- DSearch.validateRequest transporterId dSearchReq
+    fork "search request processing" $
+      Redis.whenWithLockRedis (searchProcessingLockKey dSearchReq.messageId transporterId.getId) 60 $ do
+        dSearchRes <- DSearch.handler merchant dSearchReq
+        let callbackUrl = gateway.subscriber_url
+        case req of
```

Comment:
same here, make switch case based on flag and not on request type everywhere else